File: Leetcode/2099 - Find Subsequence of Length K With the Largest Sum/solution.py
def maxSubsequence(self, nums: List[int], k: int) -> List[int]:
    indexed_nums: List[Tuple[int, int]] = [(num, idx) for idx, num in enumerate(nums)]
    indexed_nums.sort(key=lambda x: -x[0])  # Sort by value descending
    top_k: List[Tuple[int, int]] = indexed_nums[:k]
    top_k.sort(key=lambda x: x[1])  # Sort by original index
    return [num for num, _ in top_k]

File: Leetcode/2099 - Find Subsequence of Length K With the Largest Sum/solution.ts
function maxSubsequence(nums: number[], k: number): number[] {
    return nums
        .map((num: number, index: number): [number, number] => [num, index])
        .sort((a: [number, number], b: [number, number]) => b[0] - a[0]) // Sort descending by value
        .slice(0, k) // Keep top-k elements
        .sort((a: [number, number], b: [number, number]) => a[1] - b[1]) // Sort back by original index
        .map((el: [number, number]) => el[0]); // Extract values
};